### PR TITLE
Convert Class Names to Class IDs in Object Detection Nodes

### PR DESCRIPTION
**Pull request by ongtw**
Updated conversion of class names to class IDs to be activated within the initializer of the object detection nodes.

The specific changes are:
* Refactor load class ID mapping and change class name to class ID functions into `peekingduck/utils`
* Update `ConfigLoader` and `DeclarativeLoader` classes accordingly
* Add class name to class ID conversion function into `AbstractNode` initializer
* Update unit tests accordingly.

### CODE REVIEW

**Review by liyier90 — CHANGES_REQUESTED** [review]
2 suggestions for your consideration

File: peekingduck/pipeline/nodes/node.py [hunk 1]
```diff
@@ -63,6 +64,18 @@ def __init__(
         self.config_loader = ConfigLoader(pkd_base_dir)
         self.load_node_config(config, kwargs)  # type: ignore
 
+        # For object detection nodes, convert class names to class ids, if any
+        if (
+            self.node_name == "model.yolo"
+            or self.node_name == "model.efficientdet"
+            or self.node_name == "model.yolox"
+        ):
```
[the remark this answers is not in the document]
Comment (ongtw) [reply peekingduck/pipeline/nodes/node.py]: Ah no, this was "freudian" code: started with one comparison, and I mechanically added the other two.
Made the change: much cleaner as you suggested.

File: peekingduck/utils/create_node_helper.py [hunk 2]
```diff
@@ -127,6 +134,94 @@ def get_config_and_script_paths(
...
+    elif value == ["*"]:
+        logger.info("Detecting all object classes")
+        value = [*class_id_map]
+
+    value_lc = [x.lower() if isinstance(x, str) else x for x in value]
+
+    # parse value_lc for possible class name errors
+    invalid_class_names = []
+    for class_name in value_lc:
+        if isinstance(class_name, str) and class_id_map.get(class_name, -1) < 0:
```
[the remark this answers is not in the document]
Comment (ongtw) [reply peekingduck/utils/create_node_helper.py]: Updated as recommended.

**Comment by ongtw** [comment]
Made the recommended changes.

**Review by liyier90 — APPROVED** [review]

**Pull request merged**